library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -1502,32 +1533,38 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
     // while max values are related to top-left.
     private fun computeScrollerValues(horizontal: Boolean, output: ScrollerValues) {
         @ScaledPan val currentPan = (if (horizontal) scaledPanX else scaledPanY).toInt()
-        val viewDim = (if (horizontal) mContainerWidth else mContainerHeight).toInt()
+        val containerDim = (if (horizontal) mContainerWidth else mContainerHeight).toInt()
         @ScaledPan val contentDim = (if (horizontal) mContentScaledWidth else mContentScaledHeight).toInt()
         val fix = checkPanBounds(horizontal, false).toInt()
-        if (viewDim >= contentDim) {
-            // Content is smaller, we are showing some boundary.
-            // We can't move in any direction (but we can overScroll).
-            output.minValue = currentPan + fix
-            output.startValue = currentPan
-            output.maxValue = currentPan + fix
-        } else {
-            // Content is bigger, we can move.
-            // in this case minPan + viewDim = contentDim
-            output.minValue = -(contentDim - viewDim)
-            output.startValue = currentPan
+        val alignment = if (horizontal) Alignment.getHorizontal(mAlignment) else Alignment.getVertical(mAlignment)
+        if (contentDim > containerDim) {
+            // Content is bigger. We can move between 0 and extraSpace, but since our pans
+            // are negative, we must invert the sign.
+            val extraSpace = contentDim - containerDim
+            output.minValue = -extraSpace
             output.maxValue = 0
+        } else if (alignment == Alignment.NO_VALUE
+                || alignment == Alignment.NONE_VERTICAL
+                || alignment == Alignment.NONE_HORIZONTAL) {
```

Comment:
This should also work:
`else if (alignment in setOf(Alignment.NO_VALUE, Alignment.NONE_VERTICAL, Alignment.NONE_HORIZONTAL)) {`